tests/plugins/test_koji_upload.py

```diff
@@ -0,0 +1,993 @@
+"""
+Copyright (c) 2015 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals
+
+import json
+import os
+
+try:
+    import koji
+except ImportError:
+    import inspect
+    import sys
+
+    # Find our mocked koji module
+    import tests.koji as koji
+    mock_koji_path = os.path.dirname(inspect.getfile(koji.ClientSession))
+    if mock_koji_path not in sys.path:
+        sys.path.append(os.path.dirname(mock_koji_path))
+
+    # Now load it properly, the same way the plugin will
+    del koji
+    import koji
+
+from atomic_reactor.core import DockerTasker
+from atomic_reactor.plugins.post_koji_upload import (KojiUploadLogger,
+                                                     KojiUploadPlugin)
+from atomic_reactor.plugins.post_rpmqa import PostBuildRPMqaPlugin
+from atomic_reactor.plugin import PostBuildPluginsRunner, PluginFailedException
+from atomic_reactor.inner import DockerBuildWorkflow, TagConf, PushConf
+from atomic_reactor.util import ImageName, ManifestDigest
+from atomic_reactor.source import GitSource
+from atomic_reactor.build import BuildResult
+from tests.constants import SOURCE, MOCK
+
+from flexmock import flexmock
+import pytest
+from tests.docker_mock import mock_docker
+import subprocess
+from osbs.api import OSBS
+from osbs.exceptions import OsbsException
+from six import string_types
+
+NAMESPACE = 'mynamespace'
+BUILD_ID = 'build-1'
+
+
+def noop(*args, **kwargs): return None
+
+
+# temp workaround until this API is added to osbs-client
+OSBS.create_config_map = noop
+OSBS.get_config_map = noop
+
+
+class X(object):
+    pass
+
+
+class MockedOSBS(OSBS):
+    def __init__(self):
+        self.configmap = {}
+
+        (flexmock(OSBS)
+            .should_receive('get_build_logs')
+            .with_args(BUILD_ID)
+            .and_return('build logs - \u2018 \u2017 \u2019'))
+        (flexmock(OSBS)
+            .should_receive('get_pod_for_build')
+            .with_args(BUILD_ID)
+            .and_return(MockedPodResponse()))
+        (flexmock(OSBS)
+            .should_receive('create_config_map')
+            .with_args(BUILD_ID+'-md', dict)
+            .replace_with(self.create_config_map))
+        (flexmock(OSBS)
+            .should_receive('get_config_map')
+            .with_args(BUILD_ID+'-md')
+            .replace_with(self.get_config_map))
+
+    def create_config_map(self, name, data):
+        assert isinstance(data, dict)
+        assert isinstance(name, basestring)
```

Comment:
This is causing Python3 tests to fail. Instead of using `isinstance(name, basestring)`, use `is_string_type(name)`.